snuba/cli/subscriptions.py

```diff
@@ -0,0 +1,195 @@
+import signal
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+from contextlib import closing
+from datetime import timedelta
+from typing import Optional, Sequence
+
+import click
+
+from snuba import settings
+from snuba.datasets.factory import DATASET_NAMES, enforce_table_writer, get_dataset
+from snuba.environment import setup_logging, setup_sentry
+from snuba.redis import redis_client
+from snuba.subscriptions.consumer import TickConsumer
+from snuba.subscriptions.data import PartitionId
+from snuba.subscriptions.executor import SubscriptionExecutor
+from snuba.subscriptions.scheduler import SubscriptionScheduler
+from snuba.subscriptions.store import RedisSubscriptionDataStore
+from snuba.subscriptions.worker import (
+    SubscriptionResultCodec,
+    SubscriptionWorker,
+)
+from snuba.util import create_metrics
+from snuba.utils.codecs import PassthroughCodec
+from snuba.utils.streams.batching import BatchingConsumer
+from snuba.utils.streams.kafka import (
+    CommitCodec,
+    KafkaConsumer,
+    KafkaProducer,
+    build_kafka_consumer_configuration,
+)
+from snuba.utils.streams.synchronized import SynchronizedConsumer
+from snuba.utils.streams.types import Topic
+
+
+@click.command()
+@click.option(
+    "--dataset",
+    "dataset_name",
+    default="events",
+    type=click.Choice(DATASET_NAMES),
+    help="The dataset to target",
+)
+@click.option("--topic")
+@click.option("--partitions", type=int)
+@click.option("--commit-log-topic")
+@click.option(
+    "--commit-log-group",
+    "commit_log_groups",
+    multiple=True,
+    default=["snuba-consumers"],
+)
+@click.option(
+    "--consumer-group",
+    default="snuba-subscriptions",
+    help="Consumer group use for consuming the dataset source topic.",
+)
+@click.option(
+    "--auto-offset-reset",
+    default="error",
+    type=click.Choice(["error", "earliest", "latest"]),
+    help="Kafka consumer auto offset reset.",
```

Comment:
Maybe specify that this is for the main consumer and not the commit log consumer ?

Reply:
I guess so, yeah. I've been mostly considering the commit log consumer an implementation detail that is best left as hidden as possible except when it absolutely needs to be exposed (like the topic itself, since we don't use the default.)

Reply:
I think I'd prefer to leave this off (it seems too specific with the internal implementation) unless you feel strongly that it's confusing without and/or otherwise necessary.